File: 9.palindrome-number.js
/*
 * @lc app=leetcode id=9 lang=javascript
 *
 * [9] Palindrome Number
 */

// @lc code=start
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < 0) return false;
  let rev = 0;
  let xx = x;
  while (xx) {
    rev = rev * 10 + (xx % 10);
    xx = Math.trunc(xx / 10);
  }
  return rev === x;
};
// @lc code=end

File: 9.palindrome-number.go
/*
 * @lc app=leetcode id=9 lang=golang
 *
 * [9] Palindrome Number
 */

package leetcode

// @lc code=start
func isPalindrome9(x int) bool {
	rev, xx := 0, x
	for xx > 0 {
		rev = rev*10 + xx%10
		xx = xx / 10
	}
	return rev == x
}

// @lc code=end
